docs/howto/kubernetes/gardener/kubectl.md

```diff
@@ -48,26 +73,57 @@ kubectl config view
 
 You should see something like this:
 
-```yaml
-apiVersion: v1
-clusters:
-  - cluster:
-      certificate-authority-data: DATA+OMITTED
-      server: https://api.test-cluster.p40698.staging-k8s.{{gui_domain}}
-    name: shoot--p40698--test-cluster
-contexts:
-  - context:
-      cluster: shoot--p40698--test-cluster
-      user: shoot--p40698--test-cluster-token
-    name: shoot--p40698--test-cluster
-current-context: shoot--p40698--test-cluster
-kind: Config
-preferences: { }
-users:
-  - name: shoot--p40698--test-cluster-token
-    user:
-      token: REDACTED
-```
+=== "Static kubeconfig"
+    ```yaml
+    apiVersion: v1
+    clusters:
+    - cluster:
+        certificate-authority-data: DATA+OMITTED
+        server: https://api.ghar.p43597.k8s.{{gui_domain}}
+      name: shoot--p43597--ghar
+    contexts:
+    - context:
+        cluster: shoot--p43597--ghar
+        user: shoot--p43597--ghar
+      name: shoot--p43597--ghar
+    current-context: shoot--p43597--ghar
+    kind: Config
+    preferences: {}
+    users:
+    - name: shoot--p43597--ghar
+      user:
+        token: REDACTED
+    ```
+=== "Certificate-based kubeconfig"
+    ```yaml
+    apiVersion: v1
+    clusters:
+    - cluster:
+        certificate-authority-data: DATA+OMITTED
+        server: https://api.ghar.p268.staging-k8s.{{gui_domain}}
+      name: garden-p268--ghar-external
+    - cluster:
+        certificate-authority-data: DATA+OMITTED
+        server: https://api.ghar.p268.internal.staging-k8s.{{gui_domain}}
+      name: garden-p268--ghar-internal
```

Comment:
I believe we want to use the `k8s`, not `staging-k8s` subdomain here.

And also we should explain what the difference between the "internal" and "external" cluster is.

Reply:
When I run `kubectl config view` against the dev-cloud shoot cluster, I get the `staging-k8s` subdomain. Are you suggesting I should manually change this into `k8s`?

Also, although I only have a suspicion about what those `internal` and `external` clusters refer to, when I run `kubectl config view` against the **non** dev-cloud shoot cluster, I don't see `internal` nor `external`.

In not so many words, I need more context and/or pointers before providing more details about those `internal` and `external` clusters.

Reply:
Okay, in that case let's merge this PR last, i.e. leave it open until we can verify the behavior of the production environment.

Reply:
The three other PRs (#268, #269, #270) have landed, so please double-check this behaviour in production, and then update this PR (while simultaneously rebasing it on `main`).

Reply:
After the upgrade has landed in production, no matter if we have a Kubernetes 1.26 or Kubernetes 1.27 shoot, issuing `kubectl config view` gives us something like the following:

```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: DATA+OMITTED
    server: https://api.gharnew.p43597.k8s.cleura.cloud
  name: garden-p43597--gharnew-external
- cluster:
    certificate-authority-data: DATA+OMITTED
    server: https://api.gharnew.p43597.internal.k8s.cleura.cloud
  name: garden-p43597--gharnew-internal
contexts:
- context:
    cluster: garden-p43597--gharnew-external
    user: garden-p43597--gharnew-external
  name: garden-p43597--gharnew-external
- context:
    cluster: garden-p43597--gharnew-internal
    user: garden-p43597--gharnew-external
  name: garden-p43597--gharnew-internal
current-context: garden-p43597--gharnew-external
kind: Config
preferences: {}
users:
- name: garden-p43597--gharnew-external
  user:
    client-certificate-data: DATA+OMITTED
    client-key-data: DATA+OMITTED
```

Note that I did not replace actual data with `DATA+OMITTED`. That is in the actual output, along with the `internal` and `external` literals.

Reply:
In that case, please update the PR accordingly.

Reply:
Done!